examples/snake/main.go

```diff
@@ -0,0 +1,230 @@
+			g.moveDirection = 2
+		}
+	} else if inpututil.IsKeyJustPressed(ebiten.KeyDown) || inpututil.IsKeyJustPressed(ebiten.KeyS) {
+		if g.moveDirection != 4 {
+			g.moveDirection = 3
+		}
+	} else if inpututil.IsKeyJustPressed(ebiten.KeyUp) || inpututil.IsKeyJustPressed(ebiten.KeyW) {
+		if g.moveDirection != 3 {
+			g.moveDirection = 4
+		}
+	} else if inpututil.IsKeyJustPressed(ebiten.KeyEscape) {
+		g.reset()
+	}
+
+	if g.needsToMoveSnake() {
+		if g.collidesWithSelf() || g.collidesWithWall() {
+			g.reset()
+		}
+		g.nearapple = g.nearApple()
+
+		if g.collides() {
+			g.appleX = rand.Int63n(screenWidth-20) + -(screenWidth / 2)
+			g.appleY = rand.Int63n(screenHeight-20) + -(screenHeight / 2)
+			g.scalesLoc[int64(len(g.scalesLoc))] = &Scale{
+				X: g.scalesLoc[int64(len(g.scalesLoc)-1)].X,
+				Y: g.scalesLoc[int64(len(g.scalesLoc)-1)].Y,
+			}
+			if len(g.scalesLoc) > 7 && len(g.scalesLoc) < 15 {
+				g.level = 2
+				g.speed = 4
+			} else if len(g.scalesLoc) > 14 && len(g.scalesLoc) < 21 {
+				g.level = 3
+				g.speed = 3
+			} else if len(g.scalesLoc) > 20 && len(g.scalesLoc) < 30 {
+				g.level = 4
+				g.speed = 2
+			} else if len(g.scalesLoc) > 29 {
+				g.level = 5
+				g.speed = 1
+			} else {
+				g.level = 1
+			}
+			g.score++
+			if g.bestScore < g.score {
+				g.bestScore = g.score
+			}
+		}
+
+		for i := int64(len(g.scalesLoc)) - 1; i > 0; i-- {
+			g.scalesLoc[i].X = g.scalesLoc[i-1].X
+			g.scalesLoc[i].Y = g.scalesLoc[i-1].Y
+		}
+		switch g.moveDirection {
+		case 1:
+			g.scalesLoc[0].X = g.scalesLoc[0].X - 10
+		case 2:
+			g.scalesLoc[0].X += 10
+		case 3:
+			g.scalesLoc[0].Y += 10
+		case 4:
+			g.scalesLoc[0].Y -= 10
+		}
+	}
+
+	g.updateTimer()
+
+	return nil
+}
+
+// Draw ...
+func (g *Game) Draw(screen *ebiten.Image) {
+
+	if g.moveDirection == 0 {
+		ebitenutil.DebugPrint(screen, fmt.Sprintf("Press up/down/left/right to start, M to Enable/Disable Sound"))
+	} else {
+		if g.nearapple {
+			ebitenutil.DebugPrint(screen, fmt.Sprintf("FPS: %0.2f Level: %d Score: %d Best Score: %d Near", ebiten.CurrentFPS(), g.level, g.score, g.bestScore))
+		} else {
+			ebitenutil.DebugPrint(screen, fmt.Sprintf("FPS: %0.2f Level: %d Score: %d Best Score: %d", ebiten.CurrentFPS(), g.level, g.score, g.bestScore))
+		}
+	}
+
+	for _, v := range g.scalesLoc {
+		ebitenutil.DrawRect(screen, (screenWidth/2)+float64(v.X), (screenHeight/2)+float64(v.Y), 10, 10, color.RGBA{0x80, 0xa0, 0xc0, 0xff})
+	}
+	ebitenutil.DrawRect(screen, (screenWidth/2)+float64(g.appleX), (screenHeight/2)+float64(g.appleY), 10, 10, color.RGBA{0xFF, 0x00, 0x00, 0xff})
+}
+
+// Layout ...
+func (g *Game) Layout(outsideWidth, outsideHeight int) (int, int) {
+	return screenWidth, screenHeight
+}
+
+func newGame() *Game {
+	return &Game{
+		appleX:    30,
+		appleY:    30,
+		speed:     5,
+		scalesLoc: map[int64]*Scale{0: &Scale{}},
+	}
+}
+
+func main() {
+	ebiten.SetWindowSize(screenWidth, screenHeight)
+	ebiten.SetWindowTitle("Snake")
```

Comment:
Could you use the title `Snake (Ebiten Demo)`?

Reply:
done